### PR TITLE
Add News Aggregator App

### PR DESCRIPTION
**Pull request by MaratOM**

### CODE REVIEW

File: lesson4/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const exphbs = require('express-handlebars');
+const app = express();
+
+const NewsFactory = require('./lib/news');
+
+app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (ubcent) [inline lesson4/index.js:8]: Why do u need this line twice?

File: lesson4/lib/news/services/News.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    this.listItemSelector = ''
+    this.itemTitleSelector = ''
+    this.itemDateSelector = ''
+    this.itemBodySelector = ''
+  }
+
+  async getNewsList() {
+    let news = []
+
+    await rp(`${this.newsServiceUrl}${this.newsListUrn}`)
```
Comment (ubcent) [inline lesson4/lib/news/services/News.js:17]: Why is news array mutating here? It's more suitable to resolve it and then to catch it via await

File: lesson4/lib/news/services/Newsru.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    this.newsListUrn = '/allnews'
+    this.listItemSelector = '.index-news-content'
+    this.itemTitleSelector = '.article-title'
+    this.itemDateSelector = '.article-date'
+    this.itemBodySelector = '.article-text p.maintext'
+  }
+
+  getNewsItemDate($) {
+    let date = $(this.itemDateSelector).text().trim()
+    return date.substring(date.indexOf(':') + 2, date.indexOf('|')).trim()
```
Comment (ubcent) [inline lesson4/lib/news/services/Newsru.js:16]: You should take a look at your previous PR. I have noticed the same

**Review by ubcent — APPROVED** [review]
In general, well done :wink:

File: lesson4/index.js (hunk 1 above)
Comment (MaratOM) [reply lesson4/index.js:8]: Sorry. Just overlooked it.

File: lesson4/lib/news/services/News.js (hunk 2 above)
Comment (MaratOM) [reply lesson4/lib/news/services/News.js:17]: Yes. I'll have a closer look at it.

File: lesson4/lib/news/services/Newsru.js (hunk 3 above)
Comment (MaratOM) [reply lesson4/lib/news/services/Newsru.js:16]: Yes. I just did.

**Pull request merged**